src/domain/shared/errors.ts

```diff
@@ -9,8 +9,12 @@ export const RankedErrorLevel = [ErrorLevel.Info, ErrorLevel.Warn, ErrorLevel.Cr
 export class DomainError extends Error {
   name: string
   level?: ErrorLevel
-  constructor(message?: string) {
-    super(message)
+  constructor(message?: string | unknown) {
+    if (typeof message === "string" || typeof message === "undefined") {
+      super(message)
+    } else {
+      super()
+    }
     this.name = this.constructor.name
     this.level = ErrorLevel.Info
   }
```

Comment:
@nicolasburtey @bodymindarts @dolcalmi ,

I updated the shared `DomainError` to allow messages of type `unknown`. Let me know if this is what you were talking about earlier today. So now in the codebase you can return UnknownXXXError like this:

```
  } catch (err) {
    if (err instanceof Error) return new UnknownLedgerError(err.message)
    return new UnknownLedgerError(err)
  }
```

Let me know if this looks good then I can update all over the codebase.